### ISSUE TITLE
Buggy Behavior of Time Picker - MouseMove Firing Bug in Windows (TouchPan.js)

### ISSUE THREAD

**Issue by cyfung1031** [quasarframework/quasar#10721]
https://quasar.dev/vue-components/time

Testing environment: Windows PC with Edge / Chrome (Blink Engine)

To reproduce:
1. Pointerdown at the clock number
2. Hold the pointer without releasing and waiting for the "Drag" cursor
3. Release

Expected Behavior:
Same as a click action to change to next value (hour -> minute -> second)

Buggy Behavior:
No switching among hour, minute and second.

**If the user does not perform a fast clicking (a little time holding the click), it is 50% to have such experience.**

![f8c9ce18-3d7b-4ba5-877e-e34d67b30047-00 00 25 409-00 00 56 847](https://user-images.githubusercontent.com/44498510/133838984-f5f0d126-ad4f-42da-b5d2-fac780f6cff8.gif)

**Comment by metalsadman** [comment]
can't reproduce `Microsoft Edge Version 93.0.961.52 (Official build) (64-bit)`.

**Comment by cyfung1031** [comment]
> can't reproduce `Microsoft Edge Version 93.0.961.52 (Official build) (64-bit)`.

The problem is the class `no-pointer-events--children` is added to the `<body>` 
so the click event in `div.q-time__clock.cursor-pointer.non-selectable` cannot be triggered

My PC is Windows 7 64bit.

**Comment by cyfung1031** [comment]
I have checked that this issue does not occur in v0.17
https://v0-17.quasar-framework.org/components/datetime-picker.html
```
<q-datetime-picker v-model="model" type="time" />
```

How can I get this time picker component in v2?

**Comment by cyfung1031** [comment]
I guess it is the problem in TouchPan.js
As mousemove is not triggered before mouseup, it do not trigger any event but indeed the value has already been updated.

**Comment by cyfung1031** [comment]
> can't reproduce `Microsoft Edge Version 93.0.961.52 (Official build) (64-bit)`.

@metalsadman  

### Coding Issue in TouchPan.js

I found the coding problem is in TouchPan.js


![image](https://user-images.githubusercontent.com/44498510/133885866-533fd517-cfac-4b0e-8159-2e94fc9acde9.png)

it use clientX and clientY for the movement detection of touchpan.

In Windows 7 64bit OS Blink Engine, the movement can still trigger even there is no movement.

Therefore `move( evt )` will be triggered by the native MouseEvent.
Since `absX !== absY` is always false, either `ctx.move` and `ctx.end` will not proceed.


This is confirmed as this issue disappears after adding the following coding:

```

onMounted(() => {

    document.body.addEventListener('mousedown', function(evt) {
        let p = evt.target.closest('.q-time__clock.cursor-pointer.non-selectable');
        if (p) {

            requestAnimationFrame(() => {

                document.documentElement.dispatchEvent(new MouseEvent('mousemove', {
                    clientX: evt.clientX + 1,
                    clientY: evt.clientY + 1
                }))
            })

        }

    }, true)


})

```

### Suggested Solution

**It might add some coding to skip the `move` if the clientX and clientY is the same as mousedown event.**

![amkscmsdkmck](https://user-images.githubusercontent.com/44498510/133887367-1c005ec9-c8dc-405d-a469-01e57bd06bf9.png)



CodePen Demo:

https://codepen.io/cyfung1031/pen/RwgQzNE
https://codepen.io/cyfung1031/full/RwgQzNE
![3a1c699c-118b-41a7-aeb1-32e09e2a2541](https://user-images.githubusercontent.com/44498510/133887450-2d57bb41-41ec-4900-aa06-eeebe7647def.gif)


Relevant Blink Bug Reports:
https://bugs.chromium.org/p/chromium/issues/detail?id=161464
https://bugs.chromium.org/p/chromium/issues/detail?id=721341

**Comment by pdanpdan** [comment]
The main problem is we cannot find a browser that can reproduce it.

**Comment by pdanpdan** [comment]
Because I don't have a browser to reproduce it I need someone to test at https://pdanpdan.github.io/quasar-docs/vue-components/time#basic and report if it's fixed.

**Comment by cyfung1031** [comment]
> Because I don't have a browser to reproduce it I need someone to test at https://pdanpdan.github.io/quasar-docs/vue-components/time#basic and report if it's fixed.

Yes. It works for both Edge and Chrome. The pseudo mousemove event immediately after mousedown is eliminated.
![71cc77e2-1c72-47b6-9a44-569d3ce2ae26](https://user-images.githubusercontent.com/44498510/133909255-fd83f591-37fd-4588-a15b-86fd90413090.gif)

According to [the bug reports in chromium.org](https://bugs.chromium.org/p/chromium/issues/detail?id=721341), this bug would only occur in Windows OS but not in every Windows machine.
They just close the file because reproduction is limited to some machines....

As for record, my computer environment is 
 - Windows 7 SP1 x64 Ultimate 6.1.7601 SP1
 - Microsoft Edge 93.0.961.52 (Official Build) x64
 - Google Chrome 93.0.4577.82 (Official Build) x64

**Comment by rstoenescu** [comment]
@cyfung1031 Thanks for your great contribution (reporting and also PRing)!
And thanks to all participants!

Pushed PR and one more commit to polish some things.
Fix will be available in Quasar v2.1.4
